### ISSUE TITLE
meteor android crosswalk

### ISSUE THREAD

**Issue by interfaith** [meteor/meteor#3076]
A crosswalk "CordovaLib" is set in place and eclipse used to build the app.   The app will
connect  to the server but will not execute javascripts. If an app is made outside of meteor to simply access the meteor server all is well.

Simply  placing the new "CordovaLib" in the new 'override' directory will not compile via 'ant debug'.

However using eclipse, the app compiles without error, yet only the main page is rendered so  a simple
static app will work.

**Comment by crapthings** [comment]
this is the error log.

the error can solve if i use  "cordova build android" then  "ant debug" , but after  adb install apk  the app crashed before luanchscreen.

-compile:
    [javac] Compiling 29 source files to /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/bin/classes
    [javac] warning: [options] source value 1.5 is obsolete and will be removed in a future release
    [javac] warning: [options] target value 1.5 is obsolete and will be removed in a future release
    [javac] warning: [options] To suppress warnings about obsolete options, use -Xlint:-options.
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/com/idxlug41kpwnac1acgxo/xwtest.java:25: error: cannot find symbol
    [javac] public class xwtest extends CordovaActivity
    [javac]                             ^
    [javac]   symbol: class CordovaActivity
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/com/meteor/cordova/updater/CordovaUpdatePlugin.java:9: error: cannot find symbol
    [javac] import org.apache.cordova.CallbackContext;
    [javac]                          ^
    [javac]   symbol:   class CallbackContext
    [javac]   location: package org.apache.cordova
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/com/meteor/cordova/updater/CordovaUpdatePlugin.java:10: error: cannot find symbol
    [javac] import org.apache.cordova.CordovaPlugin;
    [javac]                          ^
    [javac]   symbol:   class CordovaPlugin
    [javac]   location: package org.apache.cordova
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/com/meteor/cordova/updater/CordovaUpdatePlugin.java:11: error: cannot find symbol
    [javac] import org.apache.cordova.PluginResult;
    [javac]                          ^
    [javac]   symbol:   class PluginResult
    [javac]   location: package org.apache.cordova
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/com/meteor/cordova/updater/CordovaUpdatePlugin.java:12: error: package org.apache.cordova.PluginResult does not exist
    [javac] import org.apache.cordova.PluginResult.Status;
    [javac]                                       ^
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/com/meteor/cordova/updater/CordovaUpdatePlugin.java:23: error: cannot find symbol
    [javac] public class CordovaUpdatePlugin extends CordovaPlugin {
    [javac]                                          ^
    [javac]   symbol: class CordovaPlugin
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/com/meteor/cordova/updater/CordovaUpdatePlugin.java:133: error: cannot find symbol
    [javac]     public boolean execute(String action, JSONArray inputs, CallbackContext callbackContext) throws JSONException {
    [javac]                                                             ^
    [javac]   symbol:   class CallbackContext
    [javac]   location: class CordovaUpdatePlugin
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/com/meteor/cordova/updater/CordovaUpdatePlugin.java:177: error: cannot find symbol
    [javac]     private void setLocalPath(String wwwRoot, CallbackContext callbackContext) {
    [javac]                                               ^
    [javac]   symbol:   class CallbackContext
    [javac]   location: class CordovaUpdatePlugin
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/com/meteor/cordova/updater/CordovaUpdatePlugin.java:273: error: cannot find symbol
    [javac]     private String getCordovajsRoot(CallbackContext callbackContext) {
    [javac]                                     ^
    [javac]   symbol:   class CallbackContext
    [javac]   location: class CordovaUpdatePlugin
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/com/meteor/cordova/updater/CordovaUpdatePlugin.java:285: error: cannot find symbol
    [javac]     private String startServer(String wwwRoot, String cordovaRoot, CallbackContext callbackContext)
    [javac]                                                                    ^
    [javac]   symbol:   class CallbackContext
    [javac]   location: class CordovaUpdatePlugin
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/device/Device.java:23: error: cannot find symbol
    [javac] import org.apache.cordova.CordovaWebView;
    [javac]                          ^
    [javac]   symbol:   class CordovaWebView
    [javac]   location: package org.apache.cordova
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/device/Device.java:24: error: cannot find symbol
    [javac] import org.apache.cordova.CallbackContext;
    [javac]                          ^
    [javac]   symbol:   class CallbackContext
    [javac]   location: package org.apache.cordova
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/device/Device.java:25: error: cannot find symbol
    [javac] import org.apache.cordova.CordovaPlugin;
    [javac]                          ^
    [javac]   symbol:   class CordovaPlugin
    [javac]   location: package org.apache.cordova
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/device/Device.java:26: error: cannot find symbol
    [javac] import org.apache.cordova.CordovaInterface;
    [javac]                          ^
    [javac]   symbol:   class CordovaInterface
    [javac]   location: package org.apache.cordova
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/device/Device.java:33: error: cannot find symbol
    [javac] public class Device extends CordovaPlugin {
    [javac]                             ^
    [javac]   symbol: class CordovaPlugin
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/device/Device.java:56: error: cannot find symbol
    [javac]     public void initialize(CordovaInterface cordova, CordovaWebView webView) {
    [javac]                            ^
    [javac]   symbol:   class CordovaInterface
    [javac]   location: class Device
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/device/Device.java:56: error: cannot find symbol
    [javac]     public void initialize(CordovaInterface cordova, CordovaWebView webView) {
    [javac]                                                      ^
    [javac]   symbol:   class CordovaWebView
    [javac]   location: class Device
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/device/Device.java:69: error: cannot find symbol
    [javac]     public boolean execute(String action, JSONArray args, CallbackContext callbackContext) throws JSONException {
    [javac]                                                           ^
    [javac]   symbol:   class CallbackContext
    [javac]   location: class Device
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/file/ContentFilesystem.java:30: error: cannot find symbol
    [javac] import org.apache.cordova.CordovaInterface;
    [javac]                          ^
    [javac]   symbol:   class CordovaInterface
    [javac]   location: package org.apache.cordova
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/file/ContentFilesystem.java:31: error: cannot find symbol
    [javac] import org.apache.cordova.CordovaResourceApi;
    [javac]                          ^
    [javac]   symbol:   class CordovaResourceApi
    [javac]   location: package org.apache.cordova
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/file/ContentFilesystem.java:32: error: cannot find symbol
    [javac] import org.apache.cordova.CordovaWebView;
    [javac]                          ^
    [javac]   symbol:   class CordovaWebView
    [javac]   location: package org.apache.cordova
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/file/ContentFilesystem.java:33: error: cannot find symbol
    [javac] import org.apache.cordova.PluginManager;
    [javac]                          ^
    [javac]   symbol:   class PluginManager
    [javac]   location: package org.apache.cordova
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/file/ContentFilesystem.java:46: error: cannot find symbol
    [javac] 	private CordovaInterface cordova;
    [javac] 	        ^
    [javac]   symbol:   class CordovaInterface
    [javac]   location: class ContentFilesystem
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/file/ContentFilesystem.java:47: error: cannot find symbol
    [javac] 	private CordovaResourceApi resourceApi;
    [javac] 	        ^
    [javac]   symbol:   class CordovaResourceApi
    [javac]   location: class ContentFilesystem
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/file/ContentFilesystem.java:49: error: cannot find symbol
    [javac] 	public ContentFilesystem(String name, CordovaInterface cordova, CordovaWebView webView) {
    [javac] 	                                      ^
    [javac]   symbol:   class CordovaInterface
    [javac]   location: class ContentFilesystem
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/file/ContentFilesystem.java:49: error: cannot find symbol
    [javac] 	public ContentFilesystem(String name, CordovaInterface cordova, CordovaWebView webView) {
    [javac] 	                                                                ^
    [javac]   symbol:   class CordovaWebView
    [javac]   location: class ContentFilesystem
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/file/FileHelper.java:25: error: cannot find symbol
    [javac] import org.apache.cordova.CordovaInterface;
    [javac]                          ^
    [javac]   symbol:   class CordovaInterface
    [javac]   location: package org.apache.cordova
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/file/FileHelper.java:26: error: cannot find symbol
    [javac] import org.apache.cordova.LOG;
    [javac]                          ^
    [javac]   symbol:   class LOG
    [javac]   location: package org.apache.cordova
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/file/FileHelper.java:46: error: cannot find symbol
    [javac]     public static String getRealPath(String uriString, CordovaInterface cordova) {
    [javac]                                                        ^
    [javac]   symbol:   class CordovaInterface
    [javac]   location: class FileHelper
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/file/FileHelper.java:79: error: cannot find symbol
    [javac]     public static String getRealPath(Uri uri, CordovaInterface cordova) {
    [javac]                                               ^
    [javac]   symbol:   class CordovaInterface
    [javac]   location: class FileHelper
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/file/FileHelper.java:91: error: cannot find symbol
    [javac]     public static InputStream getInputStreamFromUriString(String uriString, CordovaInterface cordova) throws IOException {
    [javac]                                                                             ^
    [javac]   symbol:   class CordovaInterface
    [javac]   location: class FileHelper
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/file/FileHelper.java:146: error: cannot find symbol
    [javac]     public static String getMimeType(String uriString, CordovaInterface cordova) {
    [javac]                                                        ^
    [javac]   symbol:   class CordovaInterface
    [javac]   location: class FileHelper
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/file/FileUtils.java:28: error: cannot find symbol
    [javac] import org.apache.cordova.CallbackContext;
    [javac]                          ^
    [javac]   symbol:   class CallbackContext
    [javac]   location: package org.apache.cordova
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/file/FileUtils.java:29: error: cannot find symbol
    [javac] import org.apache.cordova.CordovaInterface;
    [javac]                          ^
    [javac]   symbol:   class CordovaInterface
    [javac]   location: package org.apache.cordova
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/file/FileUtils.java:30: error: cannot find symbol
    [javac] import org.apache.cordova.CordovaPlugin;
    [javac]                          ^
    [javac]   symbol:   class CordovaPlugin
    [javac]   location: package org.apache.cordova
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/file/FileUtils.java:31: error: cannot find symbol
    [javac] import org.apache.cordova.CordovaWebView;
    [javac]                          ^
    [javac]   symbol:   class CordovaWebView
    [javac]   location: package org.apache.cordova
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/file/FileUtils.java:32: error: cannot find symbol
    [javac] import org.apache.cordova.PluginResult;
    [javac]                          ^
    [javac]   symbol:   class PluginResult
    [javac]   location: package org.apache.cordova
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/file/FileUtils.java:52: error: cannot find symbol
    [javac] public class FileUtils extends CordovaPlugin {
    [javac]                                ^
    [javac]   symbol: class CordovaPlugin
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/file/FileUtils.java:150: error: cannot find symbol
    [javac]     public void initialize(CordovaInterface cordova, CordovaWebView webView) {
    [javac]                            ^
    [javac]   symbol:   class CordovaInterface
    [javac]   location: class FileUtils
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/file/FileUtils.java:150: error: cannot find symbol
    [javac]     public void initialize(CordovaInterface cordova, CordovaWebView webView) {
    [javac]                                                      ^
    [javac]   symbol:   class CordovaWebView
    [javac]   location: class FileUtils
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/file/FileUtils.java:246: error: cannot find symbol
    [javac]     public boolean execute(String action, final JSONArray args, final CallbackContext callbackContext) throws JSONException {
    [javac]                                                                       ^
    [javac]   symbol:   class CallbackContext
    [javac]   location: class FileUtils
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/file/FileUtils.java:536: error: cannot find symbol
    [javac]     private void threadhelper(final FileOp f, final CallbackContext callbackContext){
    [javac]                                                     ^
    [javac]   symbol:   class CallbackContext
    [javac]   location: class FileUtils
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/file/FileUtils.java:939: error: cannot find symbol
    [javac]     public void readFileAs(final String srcURLstr, final int start, final int end, final CallbackContext callbackContext, final String encoding, final int resultType) throws MalformedURLException {
    [javac]                                                                                          ^
    [javac]   symbol:   class CallbackContext
    [javac]   location: class FileUtils
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/file/LocalFilesystem.java:34: error: cannot find symbol
    [javac] import org.apache.cordova.CordovaInterface;
    [javac]                          ^
    [javac]   symbol:   class CordovaInterface
    [javac]   location: package org.apache.cordova
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/file/LocalFilesystem.java:45: error: cannot find symbol
    [javac] 	private CordovaInterface cordova;
    [javac] 	        ^
    [javac]   symbol:   class CordovaInterface
    [javac]   location: class LocalFilesystem
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/file/LocalFilesystem.java:47: error: cannot find symbol
    [javac] 	public LocalFilesystem(String name, CordovaInterface cordova, String fsRoot) {
    [javac] 	                                    ^
    [javac]   symbol:   class CordovaInterface
    [javac]   location: class LocalFilesystem
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/filetransfer/FileTransfer.java:48: error: cannot find symbol
    [javac] import org.apache.cordova.Config;
    [javac]                          ^
    [javac]   symbol:   class Config
    [javac]   location: package org.apache.cordova
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/filetransfer/FileTransfer.java:49: error: cannot find symbol
    [javac] import org.apache.cordova.CallbackContext;
    [javac]                          ^
    [javac]   symbol:   class CallbackContext
    [javac]   location: package org.apache.cordova
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/filetransfer/FileTransfer.java:50: error: cannot find symbol
    [javac] import org.apache.cordova.CordovaPlugin;
    [javac]                          ^
    [javac]   symbol:   class CordovaPlugin
    [javac]   location: package org.apache.cordova
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/filetransfer/FileTransfer.java:51: error: cannot find symbol
    [javac] import org.apache.cordova.CordovaResourceApi;
    [javac]                          ^
    [javac]   symbol:   class CordovaResourceApi
    [javac]   location: package org.apache.cordova
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/filetransfer/FileTransfer.java:52: error: package org.apache.cordova.CordovaResourceApi does not exist
    [javac] import org.apache.cordova.CordovaResourceApi.OpenForReadResult;
    [javac]                                             ^
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/filetransfer/FileTransfer.java:53: error: cannot find symbol
    [javac] import org.apache.cordova.PluginResult;
    [javac]                          ^
    [javac]   symbol:   class PluginResult
    [javac]   location: package org.apache.cordova
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/filetransfer/FileTransfer.java:64: error: cannot find symbol
    [javac] public class FileTransfer extends CordovaPlugin {
    [javac]                                   ^
    [javac]   symbol: class CordovaPlugin
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/filetransfer/FileTransfer.java:171: error: cannot find symbol
    [javac]     public boolean execute(String action, JSONArray args, final CallbackContext callbackContext) throws JSONException {
    [javac]                                                                 ^
    [javac]   symbol:   class CallbackContext
    [javac]   location: class FileTransfer
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/filetransfer/FileTransfer.java:223: error: cannot find symbol
    [javac]     private void upload(final String source, final String target, JSONArray args, CallbackContext callbackContext) throws JSONException {
    [javac]                                                                                   ^
    [javac]   symbol:   class CallbackContext
    [javac]   location: class FileTransfer
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/filetransfer/FileTransfer.java:654: error: cannot find symbol
    [javac]     private void download(final String source, final String target, JSONArray args, CallbackContext callbackContext) throws JSONException {
    [javac]                                                                                     ^
    [javac]   symbol:   class CallbackContext
    [javac]   location: class FileTransfer
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/filetransfer/FileTransfer.java:83: error: cannot find symbol
    [javac]         CallbackContext callbackContext;
    [javac]         ^
    [javac]   symbol:   class CallbackContext
    [javac]   location: class RequestContext
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/filetransfer/FileTransfer.java:86: error: cannot find symbol
    [javac]         RequestContext(String source, String target, CallbackContext callbackContext) {
    [javac]                                                      ^
    [javac]   symbol:   class CallbackContext
    [javac]   location: class RequestContext
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/filetransfer/FileTransfer.java:91: error: cannot find symbol
    [javac]         void sendPluginResult(PluginResult pluginResult) {
    [javac]                               ^
    [javac]   symbol:   class PluginResult
    [javac]   location: class RequestContext
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/splashscreen/SplashScreen.java:22: error: cannot find symbol
    [javac] import org.apache.cordova.CallbackContext;
    [javac]                          ^
    [javac]   symbol:   class CallbackContext
    [javac]   location: package org.apache.cordova
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/splashscreen/SplashScreen.java:23: error: cannot find symbol
    [javac] import org.apache.cordova.CordovaPlugin;
    [javac]                          ^
    [javac]   symbol:   class CordovaPlugin
    [javac]   location: package org.apache.cordova
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/splashscreen/SplashScreen.java:26: error: cannot find symbol
    [javac] public class SplashScreen extends CordovaPlugin {
    [javac]                                   ^
    [javac]   symbol: class CordovaPlugin
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/splashscreen/SplashScreen.java:29: error: cannot find symbol
    [javac]     public boolean execute(String action, JSONArray args, CallbackContext callbackContext) {
    [javac]                                                           ^
    [javac]   symbol:   class CallbackContext
    [javac]   location: class SplashScreen
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/statusbar/StatusBar.java:27: error: cannot find symbol
    [javac] import org.apache.cordova.CallbackContext;
    [javac]                          ^
    [javac]   symbol:   class CallbackContext
    [javac]   location: package org.apache.cordova
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/statusbar/StatusBar.java:28: error: cannot find symbol
    [javac] import org.apache.cordova.CordovaArgs;
    [javac]                          ^
    [javac]   symbol:   class CordovaArgs
    [javac]   location: package org.apache.cordova
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/statusbar/StatusBar.java:29: error: cannot find symbol
    [javac] import org.apache.cordova.CordovaInterface;
    [javac]                          ^
    [javac]   symbol:   class CordovaInterface
    [javac]   location: package org.apache.cordova
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/statusbar/StatusBar.java:30: error: cannot find symbol
    [javac] import org.apache.cordova.CordovaPlugin;
    [javac]                          ^
    [javac]   symbol:   class CordovaPlugin
    [javac]   location: package org.apache.cordova
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/statusbar/StatusBar.java:31: error: cannot find symbol
    [javac] import org.apache.cordova.CordovaWebView;
    [javac]                          ^
    [javac]   symbol:   class CordovaWebView
    [javac]   location: package org.apache.cordova
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/statusbar/StatusBar.java:32: error: cannot find symbol
    [javac] import org.apache.cordova.PluginResult;
    [javac]                          ^
    [javac]   symbol:   class PluginResult
    [javac]   location: package org.apache.cordova
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/statusbar/StatusBar.java:35: error: cannot find symbol
    [javac] public class StatusBar extends CordovaPlugin {
    [javac]                                ^
    [javac]   symbol: class CordovaPlugin
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/statusbar/StatusBar.java:46: error: cannot find symbol
    [javac]     public void initialize(final CordovaInterface cordova, CordovaWebView webView) {
    [javac]                                  ^
    [javac]   symbol:   class CordovaInterface
    [javac]   location: class StatusBar
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/statusbar/StatusBar.java:46: error: cannot find symbol
    [javac]     public void initialize(final CordovaInterface cordova, CordovaWebView webView) {
    [javac]                                                            ^
    [javac]   symbol:   class CordovaWebView
    [javac]   location: class StatusBar
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/statusbar/StatusBar.java:70: error: cannot find symbol
    [javac]     public boolean execute(String action, CordovaArgs args, final CallbackContext callbackContext) throws JSONException {
    [javac]                                           ^
    [javac]   symbol:   class CordovaArgs
    [javac]   location: class StatusBar
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/statusbar/StatusBar.java:70: error: cannot find symbol
    [javac]     public boolean execute(String action, CordovaArgs args, final CallbackContext callbackContext) throws JSONException {
    [javac]                                                                   ^
    [javac]   symbol:   class CallbackContext
    [javac]   location: class StatusBar
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/com/idxlug41kpwnac1acgxo/xwtest.java:27: error: method does not override or implement a method from a supertype
    [javac]     @Override
    [javac]     ^
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/com/idxlug41kpwnac1acgxo/xwtest.java:30: error: cannot find symbol
    [javac]         super.onCreate(savedInstanceState);
    [javac]         ^
    [javac]   symbol:   variable super
    [javac]   location: class xwtest
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/com/idxlug41kpwnac1acgxo/xwtest.java:31: error: cannot find symbol
    [javac]         super.init();
    [javac]         ^
    [javac]   symbol:   variable super
    [javac]   location: class xwtest
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/com/idxlug41kpwnac1acgxo/xwtest.java:33: error: cannot find symbol
    [javac]         super.loadUrl(Config.getStartUrl());
    [javac]                       ^
    [javac]   symbol:   variable Config
    [javac]   location: class xwtest
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/com/idxlug41kpwnac1acgxo/xwtest.java:33: error: cannot find symbol
    [javac]         super.loadUrl(Config.getStartUrl());
    [javac]         ^
    [javac]   symbol:   variable super
    [javac]   location: class xwtest
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/com/meteor/cordova/updater/CordovaUpdatePlugin.java:49: error: method does not override or implement a method from a supertype
    [javac]     @Override
    [javac]     ^
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/com/meteor/cordova/updater/CordovaUpdatePlugin.java:132: error: method does not override or implement a method from a supertype
    [javac]     @Override
    [javac]     ^
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/com/meteor/cordova/updater/CordovaUpdatePlugin.java:159: error: cannot find symbol
    [javac]                 PluginResult result = new PluginResult(Status.INVALID_ACTION);
    [javac]                 ^
    [javac]   symbol:   class PluginResult
    [javac]   location: class CordovaUpdatePlugin
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/com/meteor/cordova/updater/CordovaUpdatePlugin.java:159: error: cannot find symbol
    [javac]                 PluginResult result = new PluginResult(Status.INVALID_ACTION);
    [javac]                                           ^
    [javac]   symbol:   class PluginResult
    [javac]   location: class CordovaUpdatePlugin
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/com/meteor/cordova/updater/CordovaUpdatePlugin.java:159: error: cannot find symbol
    [javac]                 PluginResult result = new PluginResult(Status.INVALID_ACTION);
    [javac]                                                        ^
    [javac]   symbol:   variable Status
    [javac]   location: class CordovaUpdatePlugin
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/com/meteor/cordova/updater/CordovaUpdatePlugin.java:196: error: cannot find symbol
    [javac]                 Context ctx = cordova.getActivity().getApplicationContext();
    [javac]                               ^
    [javac]   symbol:   variable cordova
    [javac]   location: class CordovaUpdatePlugin
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/com/meteor/cordova/updater/CordovaUpdatePlugin.java:256: error: cannot find symbol
    [javac]             Context ctx = cordova.getActivity().getApplicationContext();
    [javac]                           ^
    [javac]   symbol:   variable cordova
    [javac]   location: class CordovaUpdatePlugin
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/device/Device.java:57: error: cannot find symbol
    [javac]         super.initialize(cordova, webView);
    [javac]         ^
    [javac]   symbol:   variable super
    [javac]   location: class Device
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/device/Device.java:109: error: cannot find symbol
    [javac]         String uuid = Settings.Secure.getString(this.cordova.getActivity().getContentResolver(), android.provider.Settings.Secure.ANDROID_ID);
    [javac]                                                     ^
    [javac]   symbol: variable cordova
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/file/ContentFilesystem.java:53: error: incompatible types: getClass cannot be converted to Class
    [javac] 		Class webViewClass = webView.getClass();
    [javac] 		                                     ^
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/file/ContentFilesystem.java:54: error: cannot find symbol
    [javac] 		PluginManager pm = null;
    [javac] 		^
    [javac]   symbol:   class PluginManager
    [javac]   location: class ContentFilesystem
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/file/ContentFilesystem.java:57: error: cannot find symbol
    [javac] 			pm = (PluginManager) gpm.invoke(webView);
    [javac] 			      ^
    [javac]   symbol:   class PluginManager
    [javac]   location: class ContentFilesystem
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/file/ContentFilesystem.java:65: error: cannot find symbol
    [javac] 				pm = (PluginManager)pmf.get(webView);
    [javac] 				      ^
    [javac]   symbol:   class PluginManager
    [javac]   location: class ContentFilesystem
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/file/ContentFilesystem.java:70: error: cannot find symbol
    [javac] 		this.resourceApi = new CordovaResourceApi(webView.getContext(), pm);
    [javac] 		                       ^
    [javac]   symbol:   class CordovaResourceApi
    [javac]   location: class ContentFilesystem
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/file/ContentFilesystem.java:206: error: package CordovaResourceApi does not exist
    [javac]             CordovaResourceApi.OpenForReadResult ofrr = resourceApi.openForRead(srcURL.URL);
    [javac]                               ^
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/file/ContentFilesystem.java:229: error: package CordovaResourceApi does not exist
    [javac] 		CordovaResourceApi.OpenForReadResult ofrr = resourceApi.openForRead(inputURL.URL);
    [javac] 		                  ^
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/file/FileHelper.java:56: error: cannot find symbol
    [javac]                 LOG.e(LOG_TAG, "Could get real path for URI string %s", uriString);
    [javac]                 ^
    [javac]   symbol:   variable LOG
    [javac]   location: class FileHelper
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/file/FileHelper.java:61: error: cannot find symbol
    [javac]                 LOG.e(LOG_TAG, "Cannot get real path for URI string %s because it is a file:///android_asset/ URI.", uriString);
    [javac]                 ^
    [javac]   symbol:   variable LOG
    [javac]   location: class FileHelper
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/file/FileUtils.java:115: error: cannot find symbol
    [javac]                         registerFilesystem(new LocalFilesystem(fsName, cordova, fsRoot));
    [javac]                                                                        ^
    [javac]   symbol:   variable cordova
    [javac]   location: class FileUtils
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/file/FileUtils.java:149: error: method does not override or implement a method from a supertype
    [javac]     @Override
    [javac]     ^
    [javac] /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/file/FileUtils.java:151: error: cannot find symbol
    [javac]     	super.initialize(cordova, webView);
    [javac]     	^
    [javac]   symbol:   variable super
    [javac]   location: class FileUtils
    [javac] Note: /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/file/DirectoryManager.java uses or overrides a deprecated API.
    [javac] Note: Recompile with -Xlint:deprecation for details.
    [javac] Note: /Users/crapthings/meteor_sandbox/xwtest/.meteor/local/cordova-build/platforms/android/src/org/apache/cordova/file/ContentFilesystem.java uses unchecked or unsafe operations.
    [javac] Note: Recompile with -Xlint:unchecked for details.
    [javac] 100 errors
    [javac] 3 warnings

BUILD FAILED
/usr/local/Cellar/android-sdk/24.0.2/tools/ant/build.xml:720: The following error occurred while executing this line:
/usr/local/Cellar/android-sdk/24.0.2/tools/ant/build.xml:734: Compile failed; see the compiler error output for details.

Total time: 9 seconds
Zhangs-MacBook-Pro:android crapthings$

**Comment by athmangude** [comment]
Faced the same issue with `100` javac errors about a month ago. Still waiting on any developments

**Comment by insaneinc01** [comment]
Hello, any resolution on the 100 javac errors please? Does building with Meteor v1.0.4.2 and Crosswalk 11 solve this issue?
Thanks

**Comment by interfaith** [comment]
i have no problem with copying the CordovaLib from HelloWorld.
then rename to CordovaLib. However there are bugs in 'crosswalk' itself , version 8 
works best from tests of up to version 13.

**Comment by andreivolt** [comment]
+1 I've been trying multiple combos of android-platform / Crosswalk, each time with various compilation errors.

@interfaith are you talking about a solution? could you describe it / post it on GH? I've tried the approach from https://github.com/DispatchMe/meteor-todos-crosswalk and even fixed the build (it would crash because of a perms issue), but then I get a blank screen and some errors with cordova.js.

**Comment by andreivolt** [comment]
Also, I've followed @Slava's instructions, with errors as well, at least with the latest Meteor version. I've used Crosswalk before successfully (around october-november, ~1.0 release), and it worked flawlessly. Crosswalk is soooo much better than the stock webview it's ridiculous, and no Crosswalk is certainly not an option for the apps I'm building, the stock webviews make feel like the IE 5/6 days.

**Comment by crapthings** [comment]
@andreivolt 
I've passed building crosswalk once without error, but it crashed on launch.
still waiting for this

**Comment by interfaith** [comment]
simply build cordova/crosswalk HelloWorld
then replace the CordovaLib in the meteor/android result with HelloWorldCordovaLib
, use the eclipse SDK to build the app and rename the lib to CordovaLib. Version 8 works best
for advanced HTML5 tricks. So use the eclipse/adt sdk to load the project and build it.
surprise, it works !

**Comment by crapthings** [comment]
Version 8 doesn't support webrtc. are there any cordova webrtc plugin ? lol

**Comment by andreivolt** [comment]
@crapthings the launch crash is due to the app lacking ACCESS_WIFI_STATE permission (check `adb logcat`), but in my experience with https://github.com/DispatchMe/meteor-todos-crosswalk, it still doesn't work

**Comment by andreivolt** [comment]
@interfaith I don't have Eclipse and I'm not really fond of installing / using it, but I guess I'll give it a try if there's anything about that setup that makes it work - but what? Anyhow, I'll try and do a write-up of the steps

Also, what are the version constraints on Cordova / Android Platform?

**Comment by andreivolt** [comment]
@interfaith would you mind listing the exact steps? I don't have any experience with Eclipse/Java and it's pretty confusing, it would be greatly appreciated!

**Comment by andreivolt** [comment]
Ok people, I've managed to get it working, and with recent Crosswalk versions too! At this point, it is to early to say if it's reliable or not, but in my case, I have no choice, the app I'm building really benefit from Crosswalk, to the point where I don't see myself using Meteor if I can't use Crosswalk with it.

[Anyhow, I've done a write-up here with the exact steps](https://gist.github.com/andreivolt/7d40211bd5ed361e844f).

Now, having spent way too much time on such a shaky solution, allow me to express my opinion on the Meteor-Crosswalk deal:

I've read posts by @Slava here and on the Meteor forums, and perhaps from other members of the Meteor group as well, stating that the stance ATM is that Crosswalk compatibility is outside the scope of the Meteor-Cordova integration. While I understand that getting it to work reliably is an issue, I strongly disagree, running Android apps with the stock webview is just awful. IE6-level awful. In my opinion, reliable integration with Crosswalk is a big success factor for Meteor as a mobile platform, in relation, say, with Famo.us which apparently has it prioritized.

I hope this comes across right, as I have a much respect and appreciation for the Meteor team and the project as a whole is a huuuuuuge step ahead, I just think this shouldn't be left in the periphery, and it's my impression reading in the various Meteor-related discussion forums.

**Comment by boustanihani** [comment]
:+1: @andreivolt

**Comment by Slava** [comment]
@andreivolt Thank you for documenting your way to compile crosswalk! I hope it helps other people in this thread.

I want to elaborate a little bit on my stance on the issue of Crosswalk:

- Android 2.3 WebView sucks, it is slow, it is buggy, it is terrible at following the latest web standards and I share the sentiment of it dragging developers down similar to IE6-8.
- Cordova doesn't officially integrate Crosswalk as a 1st class webview. That's it, afaik, you will not find any one-line commands in the Cordova cli to add Crosswalk to your app, neither you will find any good documentation on this across Cordova's website.
- Crosswalk is a pain to compile. The errors are hard to understand, it might work for me, but not for you and I don't have enough expertise in Java build tools to even evaluate how bad it is.
- Meteor doesn't have any special modifications to Cordova or Android build process that would make Crosswalk linking any different from a plain Cordova project. All the problems arising with compilation/linking of Crosswalk with Meteor would pop up on your machine if you were to do it for a vanilla Cordova project. (Open a new issue, if this is not true, we want to fix everything that we can fix)
- Straightening up all the details and getting a smooth UX of using crosswalk is a non-trivial task. [Ionic has launched their support of Crosswalk](http://ionicframework.com/blog/crosswalk-comes-to-ionic/) just a couple of month ago in beta. Look at the comments, lots of people are hitting problems (and I believe the Ionic team is good at Cordova-related things). The Famous wrapper is not out yet, I can't assess its value and stability.
- Meteor is low on engineering resources. There are dozens of projects in Meteor and Mobile is only a part of it. It is important strategically, it is a very useful feature for a lot of our users, but we got to make decisions on what projects get the most attention. Right now, nobody at MDG is working on the Cordova support full-time. I do work on some maintenance and updates from time to time, but the main focus is on other projects. Adding Crosswalk support to the workload is just not something we can afford as of now.

I am happy to see the community to make progress here. I am willing to review PRs and small patches if they help to get Crosswalk running better and smoother. I am sure, a lot of you are following changes in the Cordova world much closer than I am.

**Comment by interfaith** [comment]
I managed to puzzle thru git commands to place the eclipse build on github. yeah !
https://github.com/interfaith/crosswalk-for-cordova/tree/master/project
This has crosswalk-arm-12/ CordovaLib copied into the meteor/android build.
A few xml files were removed that caused errors in the build. It builds clean
and attempts to load localhost:3000, which will be your meteor default project result.
Hopefully this is enough for others to get the crosswalk ball rolling. I found other html5
issues with crosswalk, though version 8 appears to work for cordova/arm.

**Comment by andreivolt** [comment]
Thank you for the quick and thorough reply Slava!

A few comments on your points:

> Cordova doesn't officially integrate Crosswalk as a 1st class webview. That's it, afaik, you will not find any one-line commands in the Cordova cli to add Crosswalk to your app, neither you will find any good documentation on this across Cordova's website.

Indeed, it's seems there's no support coming from Cordova, but a quick search show some community packages that make things easier (e.g. https://www.npmjs.com/package/cordova-android-crosswalk). Also, the Crosswalk project itself documents steps to do this.

> Crosswalk is a pain to compile. The errors are hard to understand, it might work for me, but not for you and I don't have enough expertise in Java build tools to even evaluate how bad it is.

Yes, I completely agree, and frankly, after all the hassle I went trough to get it working, I can both see how it can be a major turn off for many devs, and I would like to prevent others from having to go trough that. Which leads me to your next point...

> Meteor doesn't have any special modifications to Cordova or Android build process that would make Crosswalk linking any different from a plain Cordova project. All the problems arising with compilation/linking of Crosswalk with Meteor would pop up on your machine if you were to do it for a vanilla Cordova project. (Open a new issue, if this is not true, we want to fix everything that we can fix)

Okay, that's what I thought. The issue is ATM the Meteor Cordova linking process isn't very well documented, I haven't really dug into it yet but there must be something that causes it to work trough Eclipse while failing from the Meteor CLI. Back in october when I got it working the first time, all it took was copying the Crosswalk modified CordovaLib to `cordova-build-override`.

> It is important strategically, it is a very useful feature for a lot of our users, but we got to make decisions on what projects get the most attention.

This is were I sort of still disagree. If it's important strategically, it should be a priority. I mean, there are certainly many people building Meteor apps designed to run in a mobile context, part of those will want to publish the apps on Google Play, which means they will be invariably be unpleasantly surprised by the issues with the stock webview, and will either give up, or use a combination of hacks (as I did) to make it work.

> I am happy to see the community to make progress here. I am willing to review PRs and small patches if they help to get Crosswalk running better and smoother. I am sure, a lot of you are following changes in the Cordova world much closer than I am.

This is something I plan to work on a soon as possible. I'll reach out if I need any guidance on how things are set up ATM. Are you the lead person on Cordova stuff at MDG?

Cheers

**Comment by Slava** [comment]
Since we track feature-requests now in GitHub issues, let's make this the main thread about Crosswalk.

**Comment by fungilation** [comment]
> Meteor is low on engineering resources.

@Slava, when is MDG letting us test Galaxy so we can sooner throw money at you and rectify this out of resource bug? ;)

**Comment by Slava** [comment]
@fungilation this is not related to this issue. MDG has money, the engineering resources don't come out of pure dollars. You need to find, hire, motivate and coordinate smart and hardworking people. If you know such, we have the jobs page here: https://www.meteor.com/jobs

Regarding Galaxy and throwing money at us, contact @debergalis privately. This is not the right place for this.

**Comment by insaneinc01** [comment]
@andreivolt Thank you for your detailed instructions! 
After many weeks of pulling-hair, I was finally about to build crosswalk into my meteor+famo.us android app! I was able to make it work with crosswalk-cordova-12.41.296.4-arm, although there's no apparent reason why it didn't work with Crosswalk 11. 
I had to completely re-install meteor, android-sdk and eclipse during the process, but what the heck, its finally working!. Thanks again.

**Comment by Wenape** [comment]
@andreivolt It worked for me using the version specified in your guide. I just had to remove 2 other functions apart from `canGoBack()` and it worked. Thanks for the Info.

**Comment by andreivolt** [comment]
I've managed to build the app trough the usual Meteor build system, I'll post a repo with the steps, including the `cordova-build-override` dir with the hacks discussed above.

All in all, it works surprisingly well, except for the "splashscreen" hack with which is actually [a Crosswalk bug](https://crosswalk-project.org/jira/si/jira.issueviews:issue-html/XWALK-3612/XWALK-3612.html).

Still, I wish there was a way to integrate this in the core Meteor distribution, perhaps as a cli flag to the build / run commands, so that a compatible Crosswalk version would be automatically downloaded, hacks applied, etc., still using the overrides mechanism but in a transparent way.

@Slava does this sound like something you would consider implementing? would be happy to give a stab at this if there's buy in on your side

**Comment by HedCET** [comment]
http://pt.stackoverflow.com/questions/53282/plugin-de-splashcreen-cordova-e-%C3%ADcones-n%C3%A3o-funcionam-no-crosswalk

**Comment by interfaith** [comment]
yes , simply replace that with 'true' to avoid the error.

 if (webView.canGoBack())


I found that copying both the cordova & the CordovaLib from HelloWorld
will build

after removing the xml files which flag an error.


On Sat, Mar 28, 2015 at 3:36 PM, Linto Cheeran <notifications@github.com>
wrote:

>
> http://pt.stackoverflow.com/questions/53282/plugin-de-splashcreen-cordova-e-%C3%ADcones-n%C3%A3o-funcionam-no-crosswalk
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/meteor/meteor/issues/3076#issuecomment-87324063>.
>



-- 
-----
Fight back spam! Download the Blue Frog.
http://www.bluesecurity.com/register/s?user=c3Bvd2FnZTI3NTk%3D

**Comment by icenpop** [comment]
Hey @andreivolt. It would be great if you could post the repo. Would love an easy way to use Crosswalk with my Meteor apps. :)

**Comment by nkoren** [comment]
+1, Would love to see what you've done, @andreivolt!

**Comment by andreivolt** [comment]
I've been meaning to doing something like an script or grunt plugin that would to it properly, but by popular demand I've created a repo extracting the `cordova-build-override` directory as I'm using it in my app, along with (very) brief shell summary of the steps I've taken.

Here's the link: https://github.com/andreivolt/meteor-cordova-crosswalk-demo

Please not I haven't tested it yet but since it's the exact build I'm using in my app, I don't see why it wouldn't work. I did tweak the AndroidManifest a little bit and you might want to diff that against your own because the perms might be wrong... Anyway, most of the action is not there. What I did is build Crosswalk and Cordova against Meteor's bundled Android SDK, and workaround the errors it gave me (the Splashscreen plugin issue and an dir error from Ant).

I promise I'll do something cleaner soon :). Actually, my wish is to integrate with Meteor's Isobuild system since that's the way to go and I'm hoping the Meteor team will accept it. Crosswalk is so nice and I don't any major issues, people who are going to use Crosswalk probably know that it's riskier (in theory, what could be riskier then the default webview?! :))

**Comment by Kestanous** [comment]
:+1: overall and nice work @andreivolt 
> Crosswalk is so nice and I don't any major issues, people who are going to use Crosswalk probably know that it's riskier (in theory, **what could be riskier then the default webview?!** :))

**Comment by ShawnOceanHu** [comment]
Hi guys,
### You don't need Eclipse
Simply download the [crosswalk cordova package][1], rename the framework/ to CordovaLib, and copy it to override your meteor build's CordovaLib. 
Then the usual steps following [@Slava 's guide][2].

### If you had compile errors:

* To fix the compile errors:
```
cd CordovaLib
ant clean; ant debug/build;
ln -s bin ant-build

cd xwalk_core_library
ant clean; ant debug/build;
ln -s bin ant-build
```

* To fix the "canGoBack" not found error:
simply replace
```
webView.canGoBack()
```
with
```
webView.getNavigationHistory().canGoBack()
```

[1]: https://download.01.org/crosswalk/releases/crosswalk/android/stable
[2]: https://meteor.hackpad.com/Building-Meteor-app-with-Crosswalk-kHKh4DzGxFQ

**Comment by crapthings** [comment]
@ShawnOceanHu i'll try it.

**Comment by interfaith** [comment]
However I found that crosswalk is so large the app fails to effectively
function, the advanced Html5
operations fail to function , where as chrome does work on android.
crosswalk crashes for advanced
html5 . chrome does not.

On Wed, May 6, 2015 at 2:09 AM, crapthings <notifications@github.com> wrote:

> @ShawnOceanHu <https://github.com/ShawnOceanHu> i'll try it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/meteor/meteor/issues/3076#issuecomment-99391324>.
>



-- 
-----
Fight back spam! Download the Blue Frog.
http://www.bluesecurity.com/register/s?user=c3Bvd2FnZTI3NTk%3D

**Comment by Exilz** [comment]
@ShawnOceanHu do you need to copy framework/ into cordova-build-override ? And so, which step is the first to follow is Slava's guide ? 
Or do you need to meteor build your app then make the changes ? Meaning you can't develop directly with Crosswalk.
Btw, is there a way to upgrade bundled Cordova's version on Meteor ? The last one handles xwalk as a plugin. 
Thanks.

**Comment by crapthings** [comment]
@Exilz +1 upgrade to new cordova.

**Comment by ShawnOceanHu** [comment]
Hi @Exilz , detailed instructions:

## Build With Crosswalk for Android
### Build
* Build meteor:
```
meteor build .build --server "https://server/url"
```
* Download the [crosswalk cordova package][1], rename the framework/ to CordovaLib, and copy it to override meteor build's CordovaLib. 
* Add VERSION file:
```
cp <path_to_unpacked_arm_bundle>/VERSION VERSION
```
* Add permissons to manifest:
```
<uses-permission android:name="android.permission.ACCESS_WIFI_STATE" />
<uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
```
* Clean the app:
```
cd .build/android/project
ant clean
android update project --subprojects --path . --target "android-21"
```
* Build the CrossWalk:
```
cd CordovaLib/
android update project --subprojects --path . --target "android-21"
ant debug/release
```
* Build app:
```
cd ../
ant debug/release
```
[1]: https://download.01.org/crosswalk/releases/crosswalk/android/stable

**Comment by ShawnOceanHu** [comment]
@Exilz :+1:  for upgrading to new cordova

**Comment by ShawnOceanHu** [comment]
Hi guys I have created this [gist of build instructions][1] for convenience.

[1]: https://gist.github.com/ShawnOceanHu/b74f6d03bf3d2fac4d40

**Comment by copleykj** [comment]
Hey guys, my fork of meteor has changes on the cordova5-update branch that update Cordova to 5.0.0 and add a crosswalk plugin which could be used via meteor add crosswalk.

for now you'll have to build the dev and android bundles manually and then symlink the updated packages in your package dir but it works..

Detailed instructions are at https://forums.meteor.com/t/timeline-for-cordova-update/3361/5

**Comment by copleykj** [comment]
@Slava I'm not sure what would need done from this point, but if you would like to look and give me some direction that would be great.

**Comment by Slava** [comment]
@copleykj submit a pull request with your proposed changes (or work in progress changes), we can discuss when we have a starting point.

**Comment by fungilation** [comment]
5.0 already, did they just skipped 4.0? One big item I'm waiting for is WKWebView support. @copleykj does your fork change cordova support (in Meteor) to run over http localhost instead of file://? Great fork anyways for keeping up with the latest Cordova! Kudos.

On WkWebView: https://shazronatadobe.wordpress.com/2015/03/03/wkwebview-and-apache-cordova/

**Comment by copleykj** [comment]
@fungilation 4 seemed to be brief, but 4.2.0 was what Meteor's cordova integration was using.

Apache just released cordova-android 4.0.0 which means that plugable webviews are supported in android so I'm sure that IOS isn't far behind and cordova-ios will support them as well.. When that happens you'll be able to use the cordova plugin for WkWebView. I believe it will be this one: https://github.com/apache/cordova-plugins/tree/master/wkwebview-engine

As for changing the cordova integration to use localhost.. Meteor actually starts a webserver on the device and sets up a special url http://meteor.local which the app connects to. My updates don't change that at all.

Thanks for the kind words, I'm glad my little bit of work might help someone else out.. Hopefully this can make it into the next release. The new web views really make things much nicer for us..

**Comment by fungilation** [comment]
Meteor starts webserver on mobile client? That's news to me. WKWebView would *not* work AFAIK with the way meteor currently works on Cordova, which is by file://. WKWebView currently support requires (correct me if I'm wrong) code files served by local webserver over http. See https://github.com/meteor/meteor/issues/2822

And I meant cordova-ios 4.0 not cordova 4, which hasn't hit yet?

**Comment by copleykj** [comment]
Yes, it does actually.. The web server is a Cordova plugin created by meteor. https://github.com/meteor/com.meteor.cordova-update

And just for reference it is started here: https://github.com/meteor/meteor/blob/devel/tools/client/meteor_cordova_loader.js#L57

The app client side part of the app is served via this http server at http://meteor.local

**Comment by fungilation** [comment]
Hmm, interesting. I know http://meteor.local is a local web server but this does mean issue in "stock" cordova prior to cordova-ios 4.0 doesn't apply? See https://shazronatadobe.wordpress.com/2015/03/03/wkwebview-and-apache-cordova/

Looking forward to meteor's inclusion of cordova-ios 4.0 and the WKWebView support!

(excuse the going off topic on WKWebView)

**Comment by rbabayoff** [comment]
Just published a package to support wkwebview in meteor. See [blog post](http://practicalmeteor.com/drastically-improving-the-performance-of-your-meteor-ios-app-with-wkweb%E2%80%8Bview/) for details.

**Comment by Wenape** [comment]
@rbabayoff Thanks for the package. I'm testing it right now on a iOS device, the performance is great. The only thing I noticed is a [bug](https://github.com/Telerik-Verified-Plugins/WKWebView/issues/85) with the keyboard plugin, but it seems that it has to do with the Telerik plugin directly.

**Comment by miri-am** [comment]
would it be an option to use the cordova-plugin-crosswalk-webview  plugin: https://www.npmjs.com/package/cordova-plugin-crosswalk-webview ?

**Comment by copleykj** [comment]
Crosswalk webview requires upgrading to cordova-android 4

**Comment by man2** [comment]
is it just me or crosswalk increase my apk size to around 20MB++?

**Comment by copleykj** [comment]
@man2 yes, that sounds about right.. Unfortunately the there is a size burden that comes with crosswalk. You'll need to weight the costs and benefits.

**Comment by fungilation** [comment]
At only 20mb, the client speed boost and hence less developer headache with
optimization is well worth the cost.

**Comment by Igor1201** [comment]
@copleykj, your [solution](https://github.com/meteor/meteor/issues/3076#issuecomment-102417405) is amazing, thank you very much!
Any news on merging back to devel this wonderful branch?

**Comment by interfaith** [comment]
so i run out of memory on the android phone,the app fails

On Mon, Jun 8, 2015 at 4:01 AM, Gary F <notifications@github.com> wrote:

> At only 20mb, the client speed boost and hence less developer headache with
> optimization is well worth the cost.
> On Sun, Jun 7, 2015 at 8:38 PM Kelly Copley <notifications@github.com>
> wrote:
>
> > @man2 <https://github.com/man2> yes, that sounds about right..
> > Unfortunately the there is a size burden that comes with crosswalk.
> You'll
> > need to weight the costs and benefits.
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <https://github.com/meteor/meteor/issues/3076#issuecomment-109850763>.
>
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/meteor/meteor/issues/3076#issuecomment-109896501>.
>



-- 
-----
Fight back spam! Download the Blue Frog.
http://www.bluesecurity.com/register/s?user=c3Bvd2FnZTI3NTk%3D

**Comment by copleykj** [comment]
This is merged into devel, less the crosswalk package.. The crosswalk
plugin is easier to add now that it's been added to npm. You can now just
run meteor add cordova:cordova-plugin-crosswalk-webview

**Comment by Igor1201** [comment]
Nice! I was using stable but I'll change to devel for now. Thank you again!

**Comment by boustanihani** [comment]
Why is https://github.com/crosswalk-project/cordova-plugin-crosswalk-webview not pulished on the official cordova plugins registry http://plugins.cordova.io/ ???

**Comment by athmangude** [comment]
They are moving to npm. You can find it there
https://www.npmjs.com/package/cordova-plugin-crosswalk-webview
ᐧ

Athman Gude
P.O Box 5485 - 80401, Diani Beach
Email: athmangude@gmail.com
Phone: +254 718 751697

**Comment by boustanihani** [comment]
@athmangude you mean there won't be any central place for all cordova plugins anymore ?

**Comment by athmangude** [comment]
@boustanihani looks like npm want to be package manager for everything

**Comment by boustanihani** [comment]
@athmangude this is sad and also confusing, they could at least be linked together somewhere

**Comment by omercioglu** [comment]
Hi @copleykj. I'm just getting started with meteor, but without crosswalk for me it's useless. Could you please help give me a head start with using the updated crosswalk?

I have checked out the devel branch of meteor. What to do next? Follow the steps at https://forums.meteor.com/t/timeline-for-cordova-update/3361/5 ?

**Comment by fungilation** [comment]
I wouldn't say useless, just slower.

**Comment by copleykj** [comment]
@omercioglu those steps should still work.. There is a relevant conversation at https://forums.meteor.com/t/how-to-get-crosswalk-working-with-android-step-by-step-guide/6683/4

**Comment by omercioglu** [comment]
@copleykj thanks for pointing to the right direction. Your help and work is
really very much appreciated.

**Comment by boustanihani** [comment]
This will be available in Meteor 1.2 => ```meteor add crosswalk```

Check: https://quip.com/RXFlAk9Rc2xI (Add Crosswalk with one command)

You can try the beta now: ```meteor update --release METEOR@1.2-rc.7```
https://forums.meteor.com/t/first-public-release-candidate-of-meteor-1-2/8986

**Comment by Multiply** [comment]
@boustanihani Amazing news! :+1:

**Comment by fungilation** [comment]
Awesome! Between this and WKWebView support on iOS for mobile performance
boost

**Comment by stubailo** [comment]
Going to close this awaiting 1.2 release!
